lib/index.js

```
@@ -1,4 +1,434 @@
+   * @param {Array of Trello Member Objects} memberDetailsArray
+   * @return {Array of Assigned People Objects}
+   */
+  static formatMemberDetailsArray(memberDetailsArray) {
+    return memberDetailsArray.filter(memberDetails => (
+      !!TrelloExtractor.formatMemberDetails(memberDetails)
+    ));
+  }
+
+  /**
+   * Returns Trello Member object from API
+   * @param {string} memberId Trello user id, or username
+   * @param {Bluebird Promise} A promise holding member details
+   */
+  getMemberDetails(memberId) {
+    if (!memberId) {
+      throw new Error(`Invalid member id: "${memberId}"`);
+    }
+
+    return this.trelloClient.getAsync(`/1/members/${memberId}`);
+  }
+
+  /**
+   * Retrieves and replaces idMembers with Trello user details
+   * @param {Trello Card} card
+   * @param {Bluebird Promise} A promise holding card with updated idMembers field
+   */
+  resolveMemberDetailsFromCard(card) {
+    // If the card isn't defined - return
+    if (!card) {
+      return Promise.resolve();
+    } else if (!card.idMembers || card.idMembers.length === 0) {
+      // If the card doesn't have people assigned to it, return the card as is
+      return Promise.resolve(card);
+    }
+
+    let promiseArray = [];
+
+    // Get member details for the card
+    return Promise.all(
+      card.idMembers.map(memberId => this.getMemberDetails(memberId))
+    ).then(members => {
+      // Replace idMembers with real data (replacing the original data which was just ID's)
+      card.idMembers = members;
+
+      return Promise.resolve(card);
+    });
+  }
+
+  /**
+   * Gets all cards from a Trello list, in the form of a promise.
+   * @param {Promisified Trello Object} trello
+   * @param {Trello List JSON} list
+   * @return {Bluebird Promise} A promise which returns an array of cards
+   */
+  getCardsFromTrelloList(list) {
+    if (!list) {
+      return Promise.resolve();
```

Furthermore, it might be better to throw an error if this is unexpected behaviour.